Editors/SettingsEditor.cpp

```diff
@@ -0,0 +1,110 @@
+#include "SettingsEditor.h"
+#include "MainWindow.h"
+#include "ui_SettingsEditor.h"
+
+#include "codegen/Settings.pb.h"
+
+#include <QDebug>
+#include <QPushButton>
+
+using namespace buffers::resources;
+
+Q_DECLARE_METATYPE(buffers::SystemInfo);
+
+static std::string get_combo_system_id(const QComboBox* combo) {
+  auto data = combo->currentData();
+  auto subsystem = data.value<buffers::SystemInfo>();
+  return subsystem.id();
+}
+
+SettingsEditor::SettingsEditor(ProtoModel* model, QWidget* parent)
+    : BaseEditor(model, parent), ui(new Ui::SettingsEditor) {
+  ui->setupUi(this);
+
+  QPushButton* saveButton = ui->buttonBox->button(QDialogButtonBox::Save);
+  saveButton->setIcon(QIcon(":/actions/accept.png"));
+  connect(saveButton, &QPushButton::clicked, [=]() {
+    qDebug() << "hey";
+    Settings settings;
+    auto* api = settings.mutable_api();
+    api->set_target_audio(get_combo_system_id(ui->audioCombo));
+    api->set_target_collision(get_combo_system_id(ui->collisionCombo));
+    api->set_target_compiler(get_combo_system_id(ui->compilerCombo));
+    api->set_target_graphics(get_combo_system_id(ui->graphicsCombo));
+    api->set_target_network(get_combo_system_id(ui->networkCombo));
+    api->set_target_platform(get_combo_system_id(ui->platformCombo));
+    api->set_target_widgets(get_combo_system_id(ui->widgetsCombo));
+    api->add_extensions("Paths");
+
+    qDebug() << "hi" << ui->audioCombo->currentData().toString();

```

Comment:
Well this is tacky. 😛

Reply:
Yeah, I'll remove all this temporary debug printing before this is ever merged.

Reply:
Alright: 10dd3ae97b9fe667019119934dcf90ee8158afae

It's all gone now, it's not needed now anyway.